docs/development/release.md

```diff
@@ -3,148 +3,297 @@
 
-    # Check the module is installed successfully.
-    ls $GOPATH/pkg/mod/github.com/ray-project/kuberay/
-    # [Expected output]: ray-operator@v1.0.0-rc.0
+    ```bash
+    git push upstream v1.4.0
     ```
 
-* Release rc0 Helm charts following the [instructions](../release/helm-chart.md).
+---
+
+### Step 4: Publish Release Images
+
+Trigger the [`release-image-build`](https://github.com/ray-project/kuberay/actions/workflows/image-release.yaml) workflow to build and publish the KubeRay container images.
+
+1. Navigate to the workflow page: [https://github.com/ray-project/kuberay/actions/workflows/image-release.yaml](https://github.com/ray-project/kuberay/actions/workflows/image-release.yaml)
+2. Click the **"Run workflow"** dropdown button.
+3. Set the parameters:
+    * **Use workflow from:** Select **`Tags`** and choose the tag you just pushed (e.g., **`v1.4.0`**).
+    * **Commit reference:** Leave this blank or ensure it points to the commit associated with the tag `v1.4.0`. *Usually derived from the tag.*
+    * **Desired release version:** Enter the tag name (e.g., **`v1.4.0`**). This determines the image tag.
+4. Click **"Run workflow"**.
+
+**Verification:** Monitor the workflow run. Once completed successfully, check that the corresponding image tags are available on quay.io.
+
+---
+
+### Step 5: Publish Kubectl Plugin Release
+
+Trigger the [`release-kubectl-plugin`](https://github.com/ray-project/kuberay/actions/workflows/kubectl-plugin-release.yaml) workflow to build and publish the `kubectl-ray` binary.
+
+1. Navigate to the workflow page: [https://github.com/ray-project/kuberay/actions/workflows/kubectl-plugin-release.yaml](https://github.com/ray-project/kuberay/actions/workflows/kubectl-plugin-release.yaml)
+2. Click the **"Run workflow"** dropdown button.
+3. Set the parameters:
+    * **Use workflow from:** Select **`Tags`** and choose the tag for the release (e.g., **`v1.4.0`**).
+4. Click **"Run workflow"**.
+
+**Verification:**
+
+* Monitor the workflow run for success.
+* Check the [KubeRay Releases page](https://github.com/ray-project/kuberay/releases). A draft release corresponding to the tag `v1.4.0` should have been created (or updated if it existed) with the plugin binaries attached as assets.
+* The workflow should automatically open a Pull Request in the [krew-index repository](https://github.com/kubernetes-sigs/krew-index) to update the plugin version for `kubectl krew`. Check for this PR and ensure it looks correct. It might require manual approval/merge by krew maintainers.
+
+---
 
-* Open a PR into the Ray repo updating the operator version used in the autoscaler integration test. Make any adjustments necessary for the test to pass ([example](https://github.com/ray-project/ray/pull/40918)). Make sure the test labelled [kubernetes-operator](https://buildkite.com/ray-project/oss-ci-build-pr/builds/17146#01873a69-5ccf-4c71-b06c-ae3a4dd9aecb) passes before merging.
+### Step 6: Publish KubeRay Helm Charts
```

Comment:
Please refer to `helm-chart.md` and update the documentation instead of creating a new section without referencing it. The details are important.

Reply:
helm-chart.md seems outdated. Any objections to remove it in favor of this section?

Reply:
Added a line below about helm-chart.md. I also went through it and it seems up to date.